workers/social/lib/social/models/group/index.coffee

```diff
@@ -1545,65 +1545,117 @@ module.exports = class JGroup extends Module
       if err then callback err
       else oldAddOwner.call this, target, options, callback
 
+
+  checkUserPassword: (account, password, callback) ->
+
+    unless account or password
+      return callback 'Error occured while deleting team'
+
+    account.fetchUser (err, user) ->
+      unless err
+        unless user.checkPassword password
+          return callback 'Your password didn\'t match with our records'
+        else
+          callback()
+
+
   destroy    : permit
     advanced : [
       { permission: 'edit own groups', validateWith : Validators.own }
       { permission: 'edit groups',     superadmin   : yes }
     ]
-    success  : (client, callback) ->
+    success  : (client, password, callback) ->
 
-      JName = require '../name'
+      account = client?.connection?.delegate
+      { sessionToken } = client
 
-      removeHelper = (model, err, next) ->
-        return next err  if err
-        return next()  unless model
+      @checkUserPassword account, password, (err) =>
 
-        model.remove (err) -> next err
+        return callback new KodingError err  if err
 
-      removeHelperMany = (klass, models, err, next) ->
-        return next err  if err
-        return next()    if not models or models.length < 1
+        slug = @getAt 'slug'
+        JName = require '../name'
 
-        ids = (model._id for model in models)
-        klass.remove ({ _id: { $in: ids } }), (err) -> next err
+        errors = []
 
-      async.series [
+        kallback = (label, next, err) ->
 
-        (next) =>
-          JName.one { name: @slug }, (err, name) ->
-            removeHelper name, err, next
+          errors.push { label, err }  if err
+          next()
 
-        (next) =>
-          @fetchPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+        removeHelper = (model, err, next, label) ->
 
-        (next) =>
-          @fetchDefaultPermissionSet (err, permSet) ->
-            removeHelper permSet, err, next
+          if err or not model
+            return kallback label, next, err
 
-        (next) =>
-          JInvitation = require '../invitation'
-          JInvitation.remove { groupName: @slug }, (err) ->
-            next err
+          model.remove (err) -> kallback label, next, err
 
-        (next) =>
-          ComputeProvider = require '../computeproviders/computeprovider'
-          ComputeProvider.destroyGroupResources this, -> next()
+        index = 0
 
-        (next) =>
-          JSession = require '../session'
-          @sendNotification 'GroupDestroyed', @slug, =>
-            JSession.remove { groupName: @slug }, (err) ->
-              next err
+        queue = [
 
-        (next) =>
-          @constructor.emit 'GroupDestroyed', this
-          next()
+          (next) ->
```

Comment:
this order is incorrect and the list is incomplete, please refer to "team deletion steps" docs for the correct dependency order